ProcessMaker/processmaker
# Implement PM Block Support

## Issue & Reproduction Steps
This PR introduces various enhancements as part of the Epic ticket [FOUR-7365](https://processmaker.atlassian.net/browse/FOUR-7365), focusing on adding support for PM blocks. The following changes have been made:

1. **Sidebar Menu Order:** The ability to set the order of sidebar menu items has been implemented. Based on the design specifications, the PM Blocks menu item is now positioned as the second item after the Processes menu item.

2. **Create PM Block Modal:** A new 'Create PM Block' modal has been added, which can be triggered from the Process Listing table. This modal facilitates the creation of PM blocks with ease.
 
3. **Ellipsis Menu Addition:** The ellipsis menu in the Process Listing now includes a 'Save as PM Block' option. This allows users to save a selected process as a PM block for future use.

## To Test:

1. Install the [PM Blocks package](https://github.com/ProcessMaker/package-pm-blocks/tree/develop).
2. Install [Data Sources package](https://github.com/ProcessMaker/package-data-sources/pull/280) branch `feature/FOUR-7366`
3. Verify that the PM Blocks menu appears in the sidebar navigation, positioned as the second item from the top.
4. Open the Ellipsis menu for a process and confirm that the 'Save as PM block' option is listed.
5. Selecting 'Save as PM block' should trigger a popup where you can update configurations for the PM block.

## Related Tickets & Packages
-  Epic ticket [FOUR-7365](https://processmaker.atlassian.net/browse/FOUR-7365)
-  Publish PM Blocks - [FOUR-7366](https://processmaker.atlassian.net/browse/FOUR-7366)
- [PM Block Package](https://github.com/ProcessMaker/package-pm-blocks/tree/develop)
- Publish PM Block PR -  https://github.com/ProcessMaker/package-pm-blocks/pull/2
- Data Sources PR -  https://github.com/ProcessMaker/package-data-sources/pull/280
- Decision Engine PR - https://github.com/ProcessMaker/package-decision-engine/pull/28
- Collection Package PR - https://github.com/ProcessMaker/package-collections/pull/350
-  Saved Search PR - https://github.com/ProcessMaker/package-savedsearch/pull/358

## Code Review Checklist

ci:db:clean
ci:package-pm-blocks:develop
ci:package-collections:feature/FOUR-7366
ci:package-savedsearch:feature/FOUR-7366
ci:package-data-sources:feature/FOUR-7366
ci:package-decision-engine:feature/FOUR-7366
ci:deploy


[FOUR-7365]: https://processmaker.atlassian.net/browse/FOUR-7365?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[FOUR-7365]: https://processmaker.atlassian.net/browse/FOUR-7365?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

.
